dbt_platform_helper/providers/config.py

```diff
@@ -51,18 +70,88 @@ def load_and_validate_platform_config(self, path=PLATFORM_CONFIG_FILE):
         except FileProviderException as e:
             self.io.abort_with_error(f"Error loading configuration from {path}: {e}")
 
+        self._validate_schema_version()
+
         try:
             self._validate_platform_config()
         except SchemaError as e:
             self.io.abort_with_error(f"Schema error in {path}. {e}")
 
         return self.config
 
+    def _abort_due_to_schema_version_error(self, config_description: str, action_required: str):
+        self.io.abort_with_error(
+            "\n".join(
+                [
+                    config_description,
+                    "",
+                    action_required,
+                ]
+            )
+        )
+
+    def _validate_schema_version(self):
+        platform_config_schema_version = self.config.get("schema_version")
+        platform_helper_default_version = self.config.get("default_versions", {}).get(
+            "platform-helper", ""
+        )
+        header = SCHEMA_VERSION_MESSAGE.format(
+            platform_helper_version=self.current_platform_helper_version,
+            schema_version=self.current_platform_config_schema_version,
+            config_platform_helper_version=(
+                platform_helper_default_version if platform_helper_default_version else "N/A"
+            ),
+            config_schema_version=(
+                platform_config_schema_version if platform_config_schema_version else "N/A"
+            ),
+        )
+
+        if platform_config_schema_version:
+            self._handle_schema_version_mismatch(platform_config_schema_version, header)
+        else:
+            self._handle_missing_schema_version(platform_helper_default_version, header)
+
+    def _handle_schema_version_mismatch(self, platform_config_schema_version: int, header: str):
+        if platform_config_schema_version < self.current_platform_config_schema_version:
+            self._abort_due_to_schema_version_error(
+                header,
+                "Please upgrade your platform-config.yml by running 'platform-helper config migrate'.",
+            )
+        elif platform_config_schema_version > self.current_platform_config_schema_version:
+            self._abort_due_to_schema_version_error(
+                header,
+                f"Please update your platform-helper to a version that supports schema_version: {platform_config_schema_version}.",
+            )
+        # else the schema_version is the correct one so continue.
+
+    def _handle_missing_schema_version(self, platform_helper_default_version: str, header: str):
+        version_parts = platform_helper_default_version.split(".")
```

Comment:
Can we use https://github.com/uktrade/platform-tools/blob/main/dbt_platform_helper/providers/semantic_version.py#L65 here?